src/components/PageHeader.tsx

```diff
@@ -0,0 +1,29 @@
+import React, { FC, ReactNode } from "react";
```

Comment:
Based on our discussion from last week, an alternative to implement this component is to use a compound component pattern. This gets around having to use props for passing components, and have the additional benefit of allowing more control over children components. For example, we could expose className as a input prop for `HeaderLeftContainer` to allow styling customization for it. This is better than having a single `PageHeader` component, since if we want to customize any children components then we have to keep adding to the props of `PageHeader`. That being said, the current implementation acheives the same thing as your PageHeader, so feel free to keep things as is.

```
import React, { FC, PropsWithChildren, ReactNode } from "react";

interface HeaderLeftContainerProps {
  title: ReactNode;
}
const HeaderLeftContainer: FC<PropsWithChildren<HeaderLeftContainerProps>> = ({
  title,
  children,
}) => {
  return (
    <div className="page-header__left">
      <h1 className="p-heading--4 u-no-margin--bottom">{title}</h1>
      {children}
    </div>
  );
};

const SearchContainer: FC<PropsWithChildren> = ({ children }) => {
  return (
    <div className="page-header__search margin-right u-no-margin--bottom">
      {children}
    </div>
  );
};

const BaseActionsContainer: FC<PropsWithChildren> = ({ children }) => {
  return <div className="page-header__base-actions">{children}</div>;
};

interface PageHeader extends FC<PropsWithChildren> {
  HeaderLeftContainer: FC<PropsWithChildren<HeaderLeftContainerProps>>;
  SearchContainer: FC<PropsWithChildren>;
  BaseActionsContainer: FC<PropsWithChildren>;
}
const PageHeader: PageHeader = ({ children }) => {
  return <div className="p-panel__header page-header">{children}</div>;
};

PageHeader.HeaderLeftContainer = HeaderLeftContainer;
PageHeader.SearchContainer = SearchContainer;
PageHeader.BaseActionsContainer = BaseActionsContainer;
export default PageHeader;

/** Example usage */
const Example = () => {
  return (
    <PageHeader>
      <PageHeader.HeaderLeftContainer title="title">
        <BulkActions />
        <PageHeader.HeaderLeftContainer>
          <Search />
        </PageHeader.HeaderLeftContainer>
        <PageHeader.BaseActionsContainer>
          <BaseActions />
        </PageHeader.BaseActionsContainer>
      </PageHeader.HeaderLeftContainer>
    </PageHeader>
  );
};
```

Reply:
good idea :+1: